### PR TITLE
[Feat] VirtualKubelet HA

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr enables HA active/passive virtual kubelet deployment

### CODE REVIEW

**Comment by aleoli** [comment]
Waiting for pr #2759 to be merged

File: cmd/virtual-kubelet/root/root.go [hunk 1]
```diff
@@ -102,6 +102,28 @@ func runRootCommand(ctx context.Context, c *Opts) error {
...
+			Namespace: c.TenantNamespace,
+		},
+		LeaderElectorName: c.NodeName,
+		LeaseDuration:     c.VirtualKubeletLeaseLeaseDuration,
+		RenewDeadline:     c.VirtualKubeletLeaseRenewDeadline,
+		RetryPeriod:       c.VirtualKubeletLeaseRetryPeriod,
+	}); err != nil {
+		return err
+	} else if !leader {
+		klog.Info("This virtual-kubelet is not the leader")
```
Comment (claudiolor) [inline cmd/virtual-kubelet/root/root.go:123]: Isn't here more correct logging an error? If we exit the blocking function and the virtual kubelet is not the leader maybe something went wrong, isn't it?

**Review by claudiolor — APPROVED** [review]

**Review by fra98 — APPROVED** [review]

**Pull request merged**